backend/src/services/recent-search-cars/resent-search-cars.service.ts

```diff
@@ -0,0 +1,105 @@
+import { prisma } from '@data/prisma-client';
+
+import type {
+  RecentSearchCarsResponse,
+  RecentSearchRequestDto,
+} from '@autoline/shared/common/types/types';
+
+const getRecentSearchCars = async (
+  userId: string,
+  take: number,
+): Promise<RecentSearchCarsResponse[]> => {
+  const recentSearchCars = await prisma.users_Searches_Cars.findMany({
+    take,
+    where: {
+      user_id: userId,
+    },
+    orderBy: {
+      created_at: 'desc',
+    },
+    select: {
+      model_id: true,
+      model: {
+        select: {
+          id: true,
+          name: true,
+          year_start: true,
+          year_end: true,
+          photo_urls: true,
+          description: true,
+          brand: {
+            select: {
+              name: true,
+              logo_url: true,
+            },
+          },
+          prices_ranges: {
+            select: {
+              price_start: true,
+              price_end: true,
+            },
+          },
+        },
+      },
+    },
+  });
+
+  const cars: RecentSearchCarsResponse[] = [];
+  recentSearchCars.map((car) => {
+    const data = {
+      id: car.model.id,
+      modelName: car.model.name,
+      yearStart: car.model.year_start,
+      yearEnd: car.model.year_end,
+      photoUrls: car.model.photo_urls,
+      brand: {
+        name: car.model.brand.name,
+        logoUrl: car.model.brand.logo_url,
+      },
+      pricesRanges: car.model.prices_ranges,
+      description: car.model.description,
+    } as RecentSearchCarsResponse;
+    cars.push(data);
+  });
```

Comment:
Please re-read what the `.map` does. **It returns a new array.** Having another array that we populate via push is totally redundant here.

It's also a good practice to not have any side-effects inside the immutable methods like `.map`, `.filter` and others

Reply:
I didn't notice, I took this piece of code from our project

Reply:
oh-oh... So that piece of code needs to be re-written too